File: 141-linked-list-cycle/141-linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        fastp=head
        slowp=head
        while fastp and fastp.next is not None:
            slowp=slowp.next
            fastp=fastp.next.next
            if slowp==fastp:
                return True
        return False

File: 141-linked-list-cycle/141-linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode h) {
       ListNode f = h;
       ListNode s = h;
       while( f != null &&  f.next != null){
            s=s.next;
            f=f.next.next;
       
            if(s == f){
                return true;                
            }

       }
        return false;
//        ListNode fastp = head;
//        ListNode slowp = head;
//        while( fastp != null &&  fastp.next != null){
            
//             slowp=slowp.next;
//             fastp=fastp.next.next;
       
//             if(slowp == fastp){
//                 return true;                
//             }

//        }
//         return false;
    }
}
